src/createServer.js

```diff
@@ -4,0 +1,54 @@
+const http = require('http');
+const { convertToCase } = require('./convertToCase/convertToCase');
+const { inputValidation } = require('./convertToCase/inputValidation');
+
+const createServer = () => {
+  const server = http.createServer((req, res) => {
+    try {
+      const [path, queryString] = req.url.split('?');
+      const params = new URLSearchParams(queryString);
+
+      const targetCase = params.get('toCase');
+      const originalText = path.slice(1);
+
+      const errors = inputValidation(originalText, targetCase);
+
+      if (errors.length) {
+        res.statusCode = 400;
+        res.setHeader('Content-Type', 'application/json');
+        res.end(JSON.stringify({ errors }));
+
+        return;
+      }
```

Comment:
Consider using the 'early return' pattern here to reduce the level of nesting in your code. You could return the error response immediately if there are validation errors, and then continue with the rest of the code without needing an 'else' block or indentation.